File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/view/RootView.java
package com.nerdstone.neatformcore.domain.view;

import com.nerdstone.neatformcore.domain.model.NFormViewData;
import com.nerdstone.neatformcore.domain.model.form.NFormViewProperty;
import com.nerdstone.neatformcore.views.handlers.ViewDispatcher;

import java.util.List;

public interface RootView {

    RootView initRootView();

    void addChild(NFormView nFormView);

    void addChildren(List<NFormViewProperty> viewProperties, ViewDispatcher viewDispatcher);

    List<NFormViewData> getViewsData();

    void setupView();

}

File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/view/RootView.kt
package com.nerdstone.neatformcore.domain.view

import com.nerdstone.neatformcore.domain.model.NFormViewData
import com.nerdstone.neatformcore.domain.model.NFormViewProperty
import com.nerdstone.neatformcore.views.handlers.ViewDispatcher

interface RootView {

    val viewsData: List<NFormViewData>

    fun initRootView(): RootView

    fun addChild(nFormView: NFormView)

    fun addChildren(viewProperties: List<NFormViewProperty>, viewDispatcher: ViewDispatcher)

}
